app/Providers/RouteServiceProvider.php

```diff
@@ -19,14 +23,31 @@ class RouteServiceProvider extends ServiceProvider
     /**
      * Define your route model bindings, pattern filters, etc.
      *
-     * @param  \Illuminate\Routing\Router  $router
      * @return void
      */
     public function boot()
     {
-        //
-
         parent::boot();
+
+        Route::bind('contact', function ($value) {
```

Comment:
If you noticed, I'm not using contactId as field names in the routes file. It's to allow using route-model binding. You can take a look at "Explicit Route-Model Binding" section of Laravel's documentation.

These handle all the necessary integrity checks (contact belongs to user, task belongs to contact, etc) so you don't have to do it manually in each controller method. Coupled with FormRequest classes, we can be sure that if the request reached the controller, it has valid and appropriate data. That's why the new controllers I'm adding are much more skinny and easy to read through.